apps/cli/src/commands/boot-challenger.ts

```diff
@@ -114,26 +114,72 @@ const onAssertionConfirmedCb = async (nodeNum: any, commandInstance: Vorpal.Comm
         }
     }
 
-    const assertionNode = await getAssertion(nodeNum);
-    commandInstance.log(`[${new Date().toISOString()}] Assertion data retrieved. Starting the submission process...`);
-    try {
-        await submitAssertionToReferee(
-            cachedSecretKey,
-            nodeNum,
-            assertionNode,
-            cachedSigner!.signer,
-        );
-        commandInstance.log(`[${new Date().toISOString()}] Submitted assertion: ${nodeNum}`);
-        lastAssertionTime = Date.now();
-    } catch (error) {
-        if (error && (error as Error).message && (error as Error).message.includes('execution reverted: "9"')) {
-            commandInstance.log(`[${new Date().toISOString()}] Could not submit challenge because it was already submitted`);
-            lastAssertionTime = Date.now();
-            return;
+    // Get Last Submitted Assertion Id & Time from Chain
+    const { assertionId: lastAssertionId, time } = await getLastSubmittedAssertionIdAndTime();
+
+    // Calculate the minimum time to submit an assertion
+    const minimumTimeToSubmit = time + MINIMUM_TIME_BETWEEN_ASSERTIONS;
+
+    // Check if enough time has passed that we can submit an assertion
+    if(Date.now() > minimumTimeToSubmit) {
+        let confirmDataToSubmit: string;
+
+        const assertionNode = await getAssertion(nodeNum);
+        commandInstance.log(`[${new Date().toISOString()}] Assertion data retrieved. Starting the submission process...`);
+
+        // Determine if we are submitting for one assertion or multiple
+        if(lastAssertionId + 1 !== nodeNum) {
```

Comment:
@CryptITAustria can you point me in the right direction here? I don't think this is correct as I think you mentioned the assertionId and nodeNum are not the same, correct?

Do I need to make a second RPC call to get the previous node?